public/app/features/canvas/elements/server.tsx

```diff
@@ -0,0 +1,474 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { ColorDimensionConfig, DimensionContext, ScalarDimensionConfig } from 'app/features/dimensions';
+import { ColorDimensionEditor, ScalarDimensionEditor } from 'app/features/dimensions/editors';
+
+import { CanvasElementItem, CanvasElementProps } from '../element';
+
+interface ServerConfig {
+  blinkRate?: ScalarDimensionConfig;
+  statusColor?: ColorDimensionConfig;
+  bulbColor?: ColorDimensionConfig;
+  type: ServerTypes;
+}
+
+interface ServerData {
+  blinkRate?: number;
+  statusColor?: string;
+  bulbColor?: string;
+  type: ServerTypes;
+}
+
+enum ServerTypes {
+  Single = 'Single',
+  Stack = 'Stack',
+  Database = 'Database',
+  Terminal = 'Terminal',
+}
+
+type Props = CanvasElementProps<ServerConfig, ServerData>;
+
+const ServerDisplay = ({ data }: Props) => {
+  const styles = useStyles2(getStyles);
+
+  const bulbColor = data?.bulbColor;
+  const bulbAnimation = `blink ${data?.blinkRate ? 1 / data.blinkRate : 0}s infinite step-end`;
+
+  return (
+    <svg viewBox="0 0 207.95 197.78">
+      {data?.type === ServerTypes.Stack ? (
+        <>
+          <defs>
+            <clipPath id="servera">
+              <rect
+                className={styles.pathA}
+                x="77.108"
+                y="24.362"
+                width="115.9"
+                height="197.78"
+                d="M 77.107697,24.361513 H 193.00871 V 222.14392 H 77.107697 Z"
+              />
+              <path
+                className={styles.pathB}
+                d="m26.804 19.62h217.95v206.93h-217.95zm50.304 4.7414v197.78h115.9v-197.78z"
+              />
+            </clipPath>
+          </defs>
+          <g transform="translate(-31.804 -24.362)">
+            <path
+              className={styles.outline}
+              x="36.804028"
+              y="29.620079"
+              width="197.94514"
+              height="186.92975"
+              d="m38.921 29.62h193.71a2.1167 2.1167 45 012.1167 2.1167v182.7a2.1167 2.1167 135 01-2.1167 2.1167h-193.71a2.1167 2.1167 45 01-2.1167-2.1167v-182.7a2.1167 2.1167 135 012.1167-2.1167z"
+              clipPath="url(#servera)"
+            />
+            <rect x="41.836" y="34.652" width="187.91" height="176.89" style={{ fill: data?.statusColor }} />
```

Comment:
I'm wondering if it would make more sense to color the server element itself vs. the background box behind it (keeping that transparent?) I feel like just coloring the slim background is not as visual as coloring the element it self. Thoughts?

Also just wondering what are the brackets around each element representing? do we need them if we end up coloring the server element itself vs background?
![Screenshot 2023-01-11 at 1 45 04 PM](https://user-images.githubusercontent.com/22381771/211924469-bf2232d3-261f-47f9-b7cf-f5891d034401.png)

Reply:
Been playing around with this a bit and honestly, I think it looks better coloring the element itself. Will post some examples.